exiftool.go

```diff
@@ -152,6 +166,50 @@ func (e *Exiftool) ExtractMetadata(files ...string) []FileMetadata {
 	return fms
 }
 
+// WriteMetadata writes the given metadata for each file.
+// Any errors will be saved to FileMetadata.Err
+// Note: If you're reusing an existing FileMetadata instance,
+//       you should nil the Err before passing it to WriteMetadata
+func (e *Exiftool) WriteMetadata(fileMetadata []FileMetadata) {
+	e.lock.Lock()
+	defer e.lock.Unlock()
+
+	for i, md := range fileMetadata {
+		if _, err := os.Stat(md.File); err != nil {
+			if os.IsNotExist(err) {
+				fileMetadata[i].Err = ErrNotExist
+				continue
+			}
+
+			fileMetadata[i].Err = err
+
+			continue
+		}
+
+		for k := range md.Fields {
+			v, err := md.GetString(k)
+			if err != nil {
+				fileMetadata[i].Err = err
+				continue
+			}
+			fmt.Fprintln(e.stdin, "-"+k+"="+v)
```

Comment:
This implementation does not "removes" all metadata that are not contained in the FileMetadata"'s map[string]interface{}.
That's useful when we want to update a specific metadata (just fill the map with what changes), but if I wan't to filter existing metadata (remove 'keywords' for instance), I can try to fill the map with everything but the keywords but it won't work (to delete a metadata, the key must be "empty").
What about:
- if a key and a value is specified, you write it (or overrride it) (`-K=V`)
- if a key is specified but with a nil value, you remove the targeted metadata (`-K=`)
- add a new option to delete all metadata before writing what is explicitly specified in the map

Reply:
I like the idea of syncing the `FileMetadata.Fields` to the file via `exiftool`